com.civicrm.hrjobroles/js/test/controllers/hr-job-roles-controller/hr-job-roles-controller_test.js

```diff
@@ -178,6 +180,30 @@ define([
 
                     expect(scope.validateRole(form_data)).toBe(true);
                 });
+
+                it('should not pass validation when the start date is lower than contract start date', function(){
```

Comment:
Let's have more declarative and english-like unit tests. Also put the set up code in the `beforeEach` block, leaving only the `expect` calls in the `it` method
```js
describe('when job role start date is lower than contract start date', function () {
    beforeEach(function () {
        form_data.start_date.$viewValue = '2016-05-04';
        form_data.end_date.$viewValue = '2017-05-05';
    });

    it('throws a validation error', function () {
        expect(scope.validateRole(form_data)).not.toBe(true);
    });
});

describe('when job role end date is higher than contract end date', function () {
    beforeEach(function () {
        form_data.start_date.$viewValue = '2016-06-04';
        form_data.end_date.$viewValue = '2017-05-06';
    });

    it('throws a validation error', function () {
        expect(scope.validateRole(form_data)).not.toBe(true);
    });
});
```